src/Uno.UI/Microsoft/UI/Xaml/Controls/PersonPicture/PersonPicture.cs

```diff
@@ -203,23 +203,33 @@ void UpdateIfReady()
 
 		var templateSettings = TemplateSettings;
 		templateSettings.ActualInitials = initials;
-		if (imageSrc != null)
+
+		if (imageSrc is not null)
 		{
-			var imageBrush = templateSettings.ActualImageBrush;
-			if (imageBrush == null)
+			if (templateSettings.ActualImageBrush is ImageBrush imageBrush)
 			{
-				imageBrush = new ImageBrush();
-				imageBrush.Stretch = Stretch.UniformToFill;
-				templateSettings.ActualImageBrush = imageBrush;
+				imageBrush.ImageSource = imageSrc;
+			}
+			else
+			{
+				templateSettings.ActualImageBrush = new ImageBrush()
+				{
+					ImageSource = imageSrc,
+					Stretch = Stretch.UniformToFill
+				};
 			}
-
-			imageBrush.ImageSource = imageSrc;
 		}
 		else
 		{
 			templateSettings.ActualImageBrush = null;
 		}
 
+#if __IOS__
+		if (templateSettings.ActualImageBrush is ImageBrush brush)
+		{
+			brush.ImageOpened += RefreshPhoto;
+		}
+#endif
```

Comment:
As WinUI doesn't have this code, it likely means that there is some other behavior that's different between WinUI and Uno. Do we know?

Reply:
I could not find anything. The different behaviour, if any, could be either in the `Image` or the `ImageSource`. The problem here is that for iOS the Image loading is taking longer and the code above is not blocking the UI to wait for image to complete loading before changing the VisualState which is then causing an unrecoverable error.

Tried enforcing the UI thread in different areas with no success.

cc: @Youssef1313

Reply:
Does it work in the desktop target? If so, what's making it work in that context?

Reply:
Thanks for the update. Let's go with the workaround for now.